main.py

```diff
@@ -15,8 +15,12 @@
 @client.event
 async def on_message(message):
     # we do not want the bot to reply to itself
+    # technically useless cause self is a bot too, tought fetching .bot may be more expensive?
     if message.author == client.user:
```

Comment:
No point also checking if the client is identical if you're already checking for a bot